Add health analyzer and medical scanner ECS

Fishfish458 opened the pull request:
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->

Purpose of this is both to implement Health Analyzer and as a stepping stone to implementing some changes to the cloner.

Once cloning console system and component are integrated the medical scanner should have no UI and basically just be a mostly dumb container that holds the player entity.

This should also fix #5602 by not letting any mob without a humanoid appearance component be scanned

**Screenshots**
<!-- If applicable, add screenshots to showcase your PR. If your PR is a visual change, add
screenshots or it's liable to be closed by maintainers. -->

![health analyzer](https://user-images.githubusercontent.com/47410468/155862712-bd6f8a13-cd2f-4c7e-8b77-0a9295d820f5.gif)

![scanning](https://user-images.githubusercontent.com/47410468/155862715-0a342695-7687-4152-a9eb-c4e99fa84f7f.gif)


![crafting](https://user-images.githubusercontent.com/47410468/155862708-27480ba3-aa5e-4aeb-96b8-65fc81e3a72f.gif)

![added](https://user-images.githubusercontent.com/47410468/155863229-6e8fdf1a-5db1-416d-8f0e-d1da88019fcd.gif)



**Changelog**
<!--
Here you can fill out a changelog that will automatically be added to the game when your PR is merged
There are 4 icons for changelog entries: add, remove, tweak, fix. I trust you can figure out the rest.

You can put your name after the :cl: symbol to change the name that shows in the changelog (otherwise it takes your GitHub username)
Like so: :cl: PJB

Generally, only put things in changelogs that players actually care about. Stuff like "Refactored X system, no changes should be visible" shouldn't be on a changelog.

For writing actual entries, don't consider the entry type suffix (e.g. add) to be "part" of the sentence:
bad: - add: a new tool for engineers
good: - add: added a new tool for engineers
-->

:cl:
- add: Health Analyzers have now been added to various medical containers and the autolathe

metalgearsloth commented on Content.Shared/MedicalScanner/SharedHealthAnalyzerComponent.cs:
```diff
@@ -0,0 +1,48 @@
+        [Serializable, NetSerializable]
+        public class HealthAnalyzerBoundUserInterfaceState : BoundUserInterfaceState
+        {
+            public readonly string? TargetName;
+            public readonly bool? IsAlive;
+            public readonly string? TotalDamage;
+            public readonly List<MobDamageGroup> DamageGroups;
+
+            public HealthAnalyzerBoundUserInterfaceState(
+                string? targetName,
+                bool? isAlive,
+                string? totalDamage,
+                List<MobDamageGroup> damageGroups)
+            {
+                TargetName = targetName;
+                IsAlive = isAlive;
+                TotalDamage = totalDamage;
+                DamageGroups = damageGroups;
+            }
+        }
```
* Target name not needed.
* * IsAlive probably not needed and can be moved to shared.  
* Realistically you could probably just have this contain an EntityUid and then have the client handle it all. This also means you don't need the mobdamagegroup and it cuts down on the state's bandwidth a lot.

Fishfish458 replied:
Going to opt for this,  only passing in an entityuid and getting the damageable and name on the client.
The only downside is that there the info isn't saved between closing and opening the UI but it's definitely a lot cleaner.

metalgearsloth commented on Content.Shared/MedicalScanner/SharedHealthAnalyzerComponent.cs:
```diff
@@ -0,0 +1,48 @@
+        }
+
+        [Serializable, NetSerializable]
+        public enum HealthAnalyzerUiKey
```
```suggestion
        public enum HealthAnalyzerUiKey : byte
```

metalgearsloth reviewed (changes requested):
Okay I did my review before I checked that client does get damageablecomponent data. For now it'll be fine until interaction prediction PR in + do_afters predicted but the state should just be handled by the client instead IMO.

Fishfish458 commented on Content.Server/Medical/MedicalScannerSystem.cs:
```diff
@@ -1,42 +1,90 @@
     internal sealed class MedicalScannerSystem : EntitySystem
     {
         [Dependency] private readonly ActionBlockerSystem _actionBlockerSystem = default!;
         [Dependency] private readonly IGameTiming _gameTiming = default!;
```
I've deleted the gametiming which is what I hope it meant, it wasn't being used.

Fishfish458 commented:
So, I've gone through and changed:
Health analyzer interface now only receives the entity uid of whoever was scanned and fills out the rest itself
added cancellation token to health analyzer.
removed on component init from health analyzer
added break on stun and break on hand change to health analyzer
removed the occupant name from the medical scanner, that'll be irrelevant with the cloning console anyhow.
changed the medical scanner to now always use the occupant sprite when it is occupied.
added a small update delay to medical scanner since it shouldn't need to be every frame.
various fixes that I stupidly missed.
